test/regression/aarch64/Syscall.cc

```diff
@@ -14,6 +14,61 @@ using Syscall = AArch64RegressionTest;
 /** The maximum size of a filesystem path. */
 static const size_t LINUX_PATH_MAX = 4096;
 
+TEST_P(Syscall, getrandom) {
+  initialHeapData_.resize(16);
+  memset(initialHeapData_.data(), -1, 8);
+
+  // The assembly will be run 2 times, filling a buffer with random bytes each
+  // run. If these buffers contain different bytes, then the getrandom is
+  // random enough
+  uint8_t* buffers[2];
+
+  for (size_t i = 0; i < 2; i++) {
+    // init buffer to record the output into
+    buffers[i] = (uint8_t*)calloc(8, sizeof(uint8_t));
+
+    RUN_AARCH64(R"(
+      # Get heap address
+      mov x0, 0
+      mov x8, 214
+      svc #0
+      
+      # getrandom(buf * = [a], buflen = 4, no flags)
+      mov x1, #8
+      ldr x0, buf
+      mov x8, #278
+      svc #0
+
+      # Move buf addr into x1 to easily read it from the test
+      ldr x1, buf
+
+      .balign 1
+      buf:  .skip   4
```

Comment:
Instead of using `buf` could you not use the heap you initialised directly, and store the initial heap address in `x1` to extract the contents afterwards?

Reply:
Seeded with clock() instead of time(NULL) as the resolution of time() is too slow (seconds) to give different random numbers on sequential syscalls